source/bolts/experimental/refraction.d

```diff
@@ -0,0 +1,733 @@
+       A string that evaluates to a symbol representing the function in the
+       local context.
+    */
+
+    string localName;
+
+    /**
+       Function name. Initial value: `__traits(identifier, fun)`.
+    */
+
+    string name;
+
+    /**
+       Return a new `Function` object with the `name` attribute set to `value`.
+    */
+
+    Function withName(string value) {
+        mixin replaceAttribute!"name";
+        return copy;
+    }
+
+    ///
+    unittest {
+        pure @nogc int answer();
+        mixin(refract!(answer, "answer").withName("ultimateAnswer").mixture);
+        static assert(
+            __traits(getAttributes, ultimateAnswer) ==
+            __traits(getAttributes, answer));
+    }
+
+    /**
+       Index of function in an arbitrary collection of functions. The index can
+       be set explicitly with `withIndex`. The `refract` overloads that operate
+       on function containers set `index` to the position of the function in
+       the set of overloaded functions in a scope. Default value: -1.
+    */
+
+    int index;
+
+    /**
+       Return a new `Function` object with the `index` attribute set to
+       `value`.
+    */
+
+    Function withIndex(int value) {
+        mixin replaceAttribute!"index";
+        return copy;
+    }
+
+    /**
+       Return type. Initial value: `bolts.experimental.refraction.ReturnType!fun`.
+    */
+
+    string returnType;
+
+    /**
+       Return a new `Function` object with the `returnType` attribute set to
+       `value`.
+    */
+
+    Function withReturnType(string value) {
+        mixin replaceAttribute!"returnType";
+        return copy;
+    }
+
+    ///
+    unittest {
+        pure int answer() { return 42; }
+        mixin(
+            refract!(answer, "answer")
+            .withName("realAnswer")
+            .withReturnType("real")
+            .mixture);
+        static assert(is(typeof(realAnswer()) == real));
+        static assert(functionAttributes!realAnswer & FunctionAttribute.pure_);
+    }
+
+    /**
+       Function parameters. Initial value: from the refracted function.
+    */
+
+    Parameter[] parameters;
+
+    /**
+       Return a new `Function` object with the parameters attribute set to
+       `value`.
+    */
+
+    Function withParameters(immutable(Parameter)[] value) {
+        mixin replaceAttribute!"parameters";
+        return copy;
+    }
+
+    ///
+    unittest {
+        int answer();
+        mixin(
+            refract!(answer, "answer")
+            .withName("answerQuestion")
+            .withParameters([ Parameter().withName("question").withType("string")])
+            .mixture);
+        int control(string);
+        static assert(is(Parameters!answerQuestion == Parameters!control));
+    }
+
+    /**
+       Return a new `Function` object with `newParameters` inserted at the
+       specified `index` in the `attributes`.
+    */
+
+    Function insertParameters(uint index, immutable(Parameter)[] newParameters...) {
```

Comment:
I might've missed this too, but i'd suggest using `with` here as well to be consistent with all the other refractor methods: 'withParametersAt(0, Parameter(...))`